docs/_extra/api-reference/hypothesis-v1.yaml

```diff
@@ -0,0 +1,123 @@
+openapi: 3.0.0
+# -----------------------------------------------------------------------------
+# API metadata
+# -----------------------------------------------------------------------------
+info:
+  title: Hypothesis API
+  version: 1.0.0
+  termsOfService: https://hypothes.is/terms-of-service
+  license:
+    name: BSD (2-Clause)
+    url: https://github.com/hypothesis/h/blob/master/LICENSE
+servers:
+  - url: https://api.hypothes.is/api
+
+# -----------------------------------------------------------------------------
+# Reusable tags for grouping operations
+# -----------------------------------------------------------------------------
+tags:
+  - name: general
+  - name: annotations
+  - name: groups
+  - name: users
+
+# -----------------------------------------------------------------------------
+# Reusable components
+# -----------------------------------------------------------------------------
+components:
+
+  # -------------------------
+  # Resusable parameters
+  # -------------------------
+  parameters:
+    GroupID:
+      name: id
+      in: path
+      required: true
+      schema:
+        description: >
+          Either an 8-character alphanumeric unique identifier generated by the service
+          or an authority-provided identifier
+        oneOf:
+          - type: string
+            description: 8-character alphanumeric group ID assigned by the service
+            example: 4j7Dko3R
+          - type: string
+            pattern: "group:[a-zA-Z0-9._\\-+!~*()']{1,1024}@.*$"
+            description: Unique identifier assigned by the group's owning authority
+            example: "group:338facc93@myauthority.net"
+    GroupExpand:
+      name: expand
+      in: query
+      required: false
+      description: >
+        One or more relations to expand for the group resource
+      schema:
+        type: array
+        items:
+          type: string
+          enum:
+            - organization
+            - scopes
+
+  # --------------------------
+  # Resusable security schemes
+  # --------------------------
+  securitySchemes:
```

Comment:
OpenAPI 3.0 uses `securitySchemes` instead of swagger's `securityDefinitions`. One of the cool new options is defining OAuth 2.0 flow; here is an example:

```
    OAuth2:
      type: oauth2
      flows:
        authorizationCode:
          authorizationUrl: https://example.com/oauth/authorize
          tokenUrl: https://example.com/oauth/token
          scopes:
            read: Grants read access
            write: Grants write access
            admin: Grants access to admin operations
```

I haven't done this yet, but would like to! [More info](https://swagger.io/docs/specification/authentication/)

Reply:
(It might prevent us from having to have a separate page with documentation of our authentication mechanisms)